### PR TITLE
Add support for Lady Bug and Cosmic Avenger

### PR DESCRIPTION
**Pull request by mdeslaur**
This branch adds support for Universal Lady Bug and Cosmic Avenger.
Both work on the same hardware, but use different roms.

### CODE REVIEW

File: libraries/CZ80Cpu/CLadybugBaseGame.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
+//
+// Copyright (c) 2019, Paul R. Swan
```
Comment (prswan) [inline libraries/CZ80Cpu/CLadybugBaseGame.cpp]: You wrote it, it's your copyright :)

File: libraries/CZ80Cpu/CLadybugBaseGame.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                                                  {NO_BANK_SWITCH, 0x00D000, 0x00D3FF, 1, 0x0F, " K6", "VidRam"}, // 2114
+                                                  {NO_BANK_SWITCH, 0x00D000, 0x00D3FF, 1, 0xF0, " J6", "VidRam"}, // 2114
+                                                  {NO_BANK_SWITCH, 0x00D400, 0x00D8FF, 1, 0x0F, " L6", "ColRam"}, // 2114
+                                                  {0}
+                                                }; // end of list
+
+//
+// RAM region is the same for all games on this board set.
+//
+static const RAM_REGION s_ramRegionByteOnly[] PROGMEM = { //                                                                                      "012", "012345"
```
Comment (prswan) [inline libraries/CZ80Cpu/CLadybugBaseGame.cpp]: NIT: The comment with the character count columns don't line up, same with the write only region below.

File: libraries/CZ80Cpu/CLadybugBaseGame.cpp [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                                                      {NO_BANK_SWITCH, 0x9002, 0xFF, "   ", "DSW0  "},
+                                                      {NO_BANK_SWITCH, 0x9003, 0xFF, "   ", "DSW1  "},
+                                                      {NO_BANK_SWITCH, 0xE000, 0xFF, "   ", "IN2   "},
+                                                      {0}
+                                                    }; // end of list
+
+//
+// Output region is the same for all versions on this board set.
+//
+static const OUTPUT_REGION s_outputRegion[] PROGMEM = { //                                   "012", "012345"
```
Comment (prswan) [inline libraries/CZ80Cpu/CLadybugBaseGame.cpp]: NIT: One character off :)

File: libraries/CZ80Cpu/CZ80ACpu.h [hunk 4]
```diff
@@ -165,6 +166,16 @@ class CZ80ACpu : public ICpu
             UINT16 *data
         );
 
+        //
+        // An implementation that has a longer wait prior
```
Comment (prswan) [inline libraries/CZ80Cpu/CZ80ACpu.h:170]: The Puckman cycle was too long a wait? Just confirming. I have a change pending that adds Crazy Kong that's needed and even longer wait hold off that Puckman.

**Review by prswan — COMMENTED** [review]
Nice! Just minor stuff and it otherwise looks good to go. Push an update with those and I'll sign it off :)
Thanks for adding,
Paul.

File: libraries/CZ80Cpu/CZ80ACpu.h (hunk 4 above)
Comment (mdeslaur) [reply libraries/CZ80Cpu/CZ80ACpu.h:170]: Yeah, it was too long. I also have a Rally-X merge proposal coming up that needed it too.

**Comment by mdeslaur** [comment]
Thanks!

**Review by prswan — APPROVED** [review]
Thx

**Pull request merged**